includes/class-adst-themes-stats-api.php

```diff
@@ -448,6 +450,107 @@ public function display_theme_average_ratings( $atts ) {
 			}
 		}
 	}
+
+	/**
+	 * Shortcode
+	 *
+	 * @param int $atts Get attributes theme Slug.
+	 * @return array $theme Get theme Details.
+	 */
+	public function display_theme_average_ratings_in_star( $atts ) {
+		$wp_theme_slug = $this->get_theme_shortcode_slug( $atts );
+		if ( '' !== $wp_theme_slug ) {
+			$api_params = array(
+				'theme'    => $wp_theme_slug,
+				'per_page' => self::$per_page,
+				'fields'   => array(
+					'homepage'       => false,
+					'description'    => false,
+					'screenshot_url' => false,
+					'rating'         => true,
+				),
+			);
+			$theme      = get_option( '_site_transient_bsf_tr_theme_info_' . $wp_theme_slug );
+			if ( '' === $theme ) {
+				return __( 'Please verify theme slug.', 'wp-themes-plugins-stats' );
+			} else {
+				if ( empty( $theme ) ) {
+					$theme = $this->bsf_tr_get_text( 'theme_information', $api_params );
+					if ( 'Please verify theme slug.' === $theme ) {
+						return __( 'Please verify theme slug.', 'wp-themes-plugins-stats' );
+					}
+					return $this->display_star_rating( $theme );
+				} else {
+					$theme = $this->bsf_delete_transient( $wp_theme_slug );
+					if ( null === $theme ) {
+						return __( 'Please verify theme slug.', 'wp-themes-plugins-stats' );
+					}
+					return $this->display_star_rating( $theme );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Display star rating of theme.
+	 *
+	 * @param array $theme to get the rating of plugin.
+	 */
+	public function display_star_rating( $theme ) {
+		$rating = $theme->rating;
+		switch ( $rating ) {
```

Comment:
This switch case can be converted to a helper function and that function can be used everywhere in the plugin.